fastplotlib/graphics/_features/_deleted.py

```diff
@@ -0,0 +1,41 @@
+from ._base import GraphicFeature, FeatureEvent
+
+
+class Deleted(GraphicFeature):
+    """
+    Used when a graphic is deleted, triggers events that can be useful to indicate this graphic has been deleted
+
+    **event pick info:**
+
+     ==================== ======================== =========================================================================
+      key                  type                     description
+     ==================== ======================== =========================================================================
+      "collection-index"   int                      the index of the graphic within the collection that triggered the event
+      "world_object"       pygfx.WorldObject        world object
+     ==================== ======================== =========================================================================
+    """
+
+    def __init__(self, parent, value: bool):
+        super(Deleted, self).__init__(parent, value)
+
+    def _set(self, value: bool):
```

Comment:
So, this will get triggered when `__del__` is called and `self.deleted = True` is set?

Reply:
How are we preventing users from if a graphic has been deleted to still having reference to the graphic and trying to say something silly like `graphic.deleted=False`? Will this throw an error if the graphic has already been deleted? I would assume so because there will be no reference.

Reply:
> So, this will get triggered when `__del__` is called and `self.deleted = True` is set?

Yes, `Graphic.__del__` sets `self.deleted = True` https://github.com/fastplotlib/fastplotlib/pull/404/files#diff-a57994ecff1cde4e05c8175265219f0b5200da144ce8733b9eb560c09bcea7ddR179 

> How are we preventing users from if a graphic has been deleted to still having reference to the graphic and trying to say something silly like `graphic.deleted=False`? Will this throw an error if the graphic has already been deleted? I would assume so because there will be no reference.

Since users are only given acccess to weakreferences, it will throw a reference error if they try to do _anything_ with a deleted `Graphic`.